tests/functional/u/undefined/undefined_variable.py

```diff
@@ -28,8 +28,8 @@ def bad_default(var, default=unknown2):  # [undefined-variable]
     """function with default arg's value set to an nonexistent name"""
     print(var, default)
     print(xxxx)  # [undefined-variable]
-    augvar += 1  # [undefined-variable]
-    del vardel  # [undefined-variable]
+    augvar += 1  # [undefined-variable, unused-variable]
+    del vardel  # [undefined-variable, unused-variable]
```

Comment:
Isn't this a false positive? Why are we reporting `unused-variable` on a `del` operation with a `undefined-variable`.

Reply:
If I understand correctly we're going to merge as it is and then open another MR to handle this comment ? I'm ok with this plan.

Reply:
I don't think so, and if @jacobtylerwalls wanted to do so I would advise against.
Nothing personal, but we never know what might come up in any of our personal lives and I don't think it's good to add code we know raises a false-positive instead of correctly warning previously. I have 100% confidence @jacobtylerwalls will get to this, but I don't think we should merge the "faulty" code.

Reply:
Addressed this in 53c5200, happy to hear your thoughts. These are statements used and defined in the same line, and we have a variable already to track that, so the approach is to just check that condition and mark as consumed.